soc: arm: stm32 with USB Type-C dead battery disabled if needed

FRASTM opened the pull request:
Check the driver configuration to disable
the USB Type-C dead battery, only if
- a USB PD system is in place (the UCPD node is enabled)
- or the user does not require USB PHY anyway

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/68577

erwango commented:
^^@JPHutchins

nordicjm reviewed (dismissed):
Needs converting to hwmv2

FRASTM committed:
soc: arm: stm32 with USB Type-C dead battery disabled if needed

Check the driver configuration to disable
the USB Type-C dead battery, only if
 - a USB PD system is in place (the UCPD node is enabled)
 - or the user does not require USB PHY anyway

Signed-off-by: Francois Ramu <francois.ramu@st.com>
